diff --git a/node_modules/@react-native-community/eslint-config/index.js b/node_modules/@react-native-community/eslint-config/index.js
index be53966..4dcb7a0 100644
--- a/node_modules/@react-native-community/eslint-config/index.js
+++ b/node_modules/@react-native-community/eslint-config/index.js
@@ -40,14 +40,6 @@ module.exports = {
     {
       files: ['*.js'],
       parser: 'babel-eslint',
-      plugins: ['flowtype'],
-      rules: {
-        // Flow Plugin
-        // The following rules are made available via `eslint-plugin-flowtype`
-
-        'flowtype/define-flow-type': 1,
-        'flowtype/use-flow-type': 1,
-      },
     },
     {
       files: ['*.ts', '*.tsx'],
