diff --git a/node_modules/@react-native-community/cli-platform-android/build/commands/runAndroid/index.js b/node_modules/@react-native-community/cli-platform-android/build/commands/runAndroid/index.js
index efeb87c..69a991d 100644
--- a/node_modules/@react-native-community/cli-platform-android/build/commands/runAndroid/index.js
+++ b/node_modules/@react-native-community/cli-platform-android/build/commands/runAndroid/index.js
@@ -160,7 +160,7 @@ function buildAndRun(args) {

   }

-  const launchPackageName = (0, _getLaunchPackageName.default)(args.variant);
+  const launchPackageName = "io.robertying.learnX";
   const adbPath = (0, _getAdbPath.default)();

   if (args.deviceId) {
