diff --git a/node_modules/@types/react-native/index.d.ts b/node_modules/@types/react-native/index.d.ts
index 58b36cb..6f2a07f 100644
--- a/node_modules/@types/react-native/index.d.ts
+++ b/node_modules/@types/react-native/index.d.ts
@@ -2297,6 +2297,9 @@ export interface AccessibilityProps extends AccessibilityPropsAndroid, Accessibi
      */
     accessibilityValue?: AccessibilityValue;
 
+    accessibilityComponentType?: any;
+    accessibilityTraits?: any;
+
     /**
      * When `accessible` is true, the system will try to invoke this function when the user performs an accessibility custom action.
      */
