diff --git a/node_modules/amazon-cognito-identity-js/src/StorageHelper-rn.js b/node_modules/amazon-cognito-identity-js/src/StorageHelper-rn.js
index 221b92a..2a98054 100644
--- a/node_modules/amazon-cognito-identity-js/src/StorageHelper-rn.js
+++ b/node_modules/amazon-cognito-identity-js/src/StorageHelper-rn.js
@@ -15,7 +15,7 @@
  * limitations under the License.
  */

-import { AsyncStorage } from 'react-native';
+import AsyncStorage from '@react-native-community/async-storage';

 const MEMORY_KEY_PREFIX = '@MemoryStorage:';
 let dataMemory = {};
