diff --git a/node_modules/appcenter/ios/appcenter-core.podspec b/node_modules/appcenter/appcenter-core.podspec
similarity index 88%
rename from node_modules/appcenter/ios/appcenter-core.podspec
rename to node_modules/appcenter/appcenter-core.podspec
index a3ce4d9..e44aef3 100644
--- a/node_modules/appcenter/ios/appcenter-core.podspec
+++ b/node_modules/appcenter/appcenter-core.podspec
@@ -1,6 +1,6 @@
 require 'json'

-package = JSON.parse(File.read(File.join(__dir__, '../package.json')))
+package = JSON.parse(File.read(File.join(__dir__, './package.json')))

 Pod::Spec.new do |s|
   # The name is hardcoded due to a name conflict, resulting in the error 'Errno::ENOENT - No such file or directory @ rb_sysopen - appcenter.podspec.json' error.
@@ -15,7 +15,7 @@ Pod::Spec.new do |s|
   s.author           = { 'Microsoft' => 'appcentersdk@microsoft.com' }

   s.source            = { :git => "https://github.com/microsoft/appcenter-sdk-react-native.git" }
-  s.source_files      = "AppCenterReactNative/**/*.{h,m}"
+  s.source_files      = "ios/AppCenterReactNative/**/*.{h,m}"
   s.platform          = :ios, '9.0'
   s.requires_arc      = true

diff --git a/node_modules/appcenter/react-native.config.js b/node_modules/appcenter/react-native.config.js
index 9d40c82..05cf021 100644
--- a/node_modules/appcenter/react-native.config.js
+++ b/node_modules/appcenter/react-native.config.js
@@ -1,11 +1,7 @@
-const path = require('path');
-
 module.exports = {
   dependency: {
     platforms: {
-      ios: {
-        podspecPath: path.join(__dirname, 'ios', 'appcenter-core.podspec')
-      },
+      ios: {},
       android: {
         packageInstance: 'new AppCenterReactNativePackage(getApplication())'
       }
