diff --git a/node_modules/expo-intent-launcher/android/build.gradle b/node_modules/expo-intent-launcher/android/build.gradle
index a464f3b..00ee934 100644
--- a/node_modules/expo-intent-launcher/android/build.gradle
+++ b/node_modules/expo-intent-launcher/android/build.gradle
@@ -1,6 +1,6 @@
 apply plugin: 'com.android.library'
 apply plugin: 'kotlin-android'
-apply plugin: 'maven'
+apply plugin: 'maven-publish'

 group = 'host.exp.exponent'
 version = '10.1.0'
@@ -20,27 +20,16 @@ buildscript {
   }
 }

-// Upload android library to maven with javadoc and android sources
-configurations {
-  deployerJars
-}
-
 // Creating sources with comments
 task androidSourcesJar(type: Jar) {
   classifier = 'sources'
   from android.sourceSets.main.java.srcDirs
 }

-// Put the androidSources and javadoc to the artifacts
-artifacts {
-  archives androidSourcesJar
-}
-
-uploadArchives {
-  repositories {
-    mavenDeployer {
-      configuration = configurations.deployerJars
-      repository(url: mavenLocal().url)
+publishing {
+  publications {
+    mavenLocal(MavenPublication) {
+      artifact androidSourcesJar
     }
   }
 }
