diff --git a/node_modules/expo-calendar/ios/CalendarModule.swift b/node_modules/expo-calendar/ios/CalendarModule.swift
index 82c1d92..96cb1b7 100644
--- a/node_modules/expo-calendar/ios/CalendarModule.swift
+++ b/node_modules/expo-calendar/ios/CalendarModule.swift
@@ -140,6 +140,9 @@ public class CalendarModule: Module {
         calendarEvent.endDate = parse(date: endDate)
       }
 
+      calendarEvent.title = event.title
+      calendarEvent.location = event.location
+      calendarEvent.notes = event.notes
       calendarEvent.isAllDay = event.allDay
       calendarEvent.availability = getAvailability(availability: event.availability)
 
@@ -258,6 +261,17 @@ public class CalendarModule: Module {
         reminder.completionDate = completionDate
       }
 
+      if let notes = details.notes {
+        reminder.notes = notes
+      }
+
+      if let isCompleted = details.completed {
+        reminder.isCompleted = isCompleted
+      }
+
+      reminder.title = details.title
+      reminder.location = details.location
+
       try eventStore.save(reminder, commit: true)
       return reminder.calendarItemIdentifier
     }
