diff --git a/node_modules/expo-file-system/android/src/main/java/expo/modules/filesystem/AndroidManifest.xml b/node_modules/expo-file-system/android/src/main/AndroidManifest.xml
similarity index 100%
rename from node_modules/expo-file-system/android/src/main/java/expo/modules/filesystem/AndroidManifest.xml
rename to node_modules/expo-file-system/android/src/main/AndroidManifest.xml
diff --git a/node_modules/expo-file-system/android/src/main/java/expo/modules/filesystem/legacy/FileSystemFileProvider.kt b/node_modules/expo-file-system/android/src/main/java/expo/modules/filesystem/FileSystemFileProvider.kt
similarity index 69%
rename from node_modules/expo-file-system/android/src/main/java/expo/modules/filesystem/legacy/FileSystemFileProvider.kt
rename to node_modules/expo-file-system/android/src/main/java/expo/modules/filesystem/FileSystemFileProvider.kt
index 011863c..52efaeb 100644
--- a/node_modules/expo-file-system/android/src/main/java/expo/modules/filesystem/legacy/FileSystemFileProvider.kt
+++ b/node_modules/expo-file-system/android/src/main/java/expo/modules/filesystem/FileSystemFileProvider.kt
@@ -1,4 +1,4 @@
-package expo.modules.filesystem.legacy
+package expo.modules.filesystem
 
 import androidx.core.content.FileProvider
 
diff --git a/node_modules/expo-file-system/expo-module.config.json b/node_modules/expo-file-system/expo-module.config.json
index 9ee0bf5..61c6eb9 100644
--- a/node_modules/expo-file-system/expo-module.config.json
+++ b/node_modules/expo-file-system/expo-module.config.json
@@ -8,12 +8,6 @@
     "modules": [
       "expo.modules.filesystem.FileSystemModule",
       "expo.modules.filesystem.legacy.FileSystemLegacyModule"
-    ],
-    "publication": {
-      "groupId": "host.exp.exponent",
-      "artifactId": "expo.modules.filesystem",
-      "version": "19.0.15",
-      "repository": "local-maven-repo"
-    }
+    ]
   }
 }
diff --git a/node_modules/expo-file-system/ios/Legacy/FileSystemHelpers.swift b/node_modules/expo-file-system/ios/Legacy/FileSystemHelpers.swift
index 11f1c7d..3101e15 100644
--- a/node_modules/expo-file-system/ios/Legacy/FileSystemHelpers.swift
+++ b/node_modules/expo-file-system/ios/Legacy/FileSystemHelpers.swift
@@ -63,6 +63,13 @@ internal func ensurePathPermission(_ appContext: AppContext?, path: String, flag
   guard let permissionsManager: EXFilePermissionModuleInterface = appContext?.legacyModule(implementing: EXFilePermissionModuleInterface.self) else {
     throw Exceptions.PermissionsModuleNotFound()
   }
+
+  if let downloads = FileManager.default.urls(for: .downloadsDirectory, in: .userDomainMask).first {
+    if path.hasPrefix(downloads.path) {
+      return
+    }
+  }
+
   guard permissionsManager.getPathPermissions(path).contains(flag) else {
     throw flag == .read ? FileNotReadableException(path) : FileNotWritableException(path)
   }
