diff --git a/node_modules/expo-notifications/build/PushTokenManager.native.js b/node_modules/expo-notifications/build/PushTokenManager.native.js
index 27bd573..e034780 100644
--- a/node_modules/expo-notifications/build/PushTokenManager.native.js
+++ b/node_modules/expo-notifications/build/PushTokenManager.native.js
@@ -1,3 +1,17 @@
-import { requireNativeModule } from 'expo-modules-core';
-export default requireNativeModule('ExpoPushTokenManager');
+import { Platform, requireNativeModule } from "expo-modules-core";
+let nativeModule = {
+  __expo_module_name__: "ExpoPushTokenManager",
+  addListener: () => {},
+  emit: () => {},
+  getDevicePushTokenAsync: () => {},
+  listenerCount: () => {},
+  removeAllListeners: () => {},
+  removeListener: () => {},
+  removeSubscription: () => {},
+  unregisterForNotificationsAsync: () => {},
+};
+if (Platform.OS === "ios") {
+  nativeModule = requireNativeModule("ExpoPushTokenManager");
+}
+export default nativeModule;
 //# sourceMappingURL=PushTokenManager.native.js.map
diff --git a/node_modules/expo-notifications/build/getDevicePushTokenAsync.js b/node_modules/expo-notifications/build/getDevicePushTokenAsync.js
index 5deaa81..6f4a56d 100644
--- a/node_modules/expo-notifications/build/getDevicePushTokenAsync.js
+++ b/node_modules/expo-notifications/build/getDevicePushTokenAsync.js
@@ -7,6 +7,9 @@ let nativeTokenPromise = null;
  * @header fetch
  */
 export default async function getDevicePushTokenAsync() {
+    if (Platform.OS === 'android') {
+        throw new Error("Push notifications are not supported on Android.");
+    }
     if (!PushTokenManager.getDevicePushTokenAsync) {
         throw new UnavailabilityError('ExpoNotifications', 'getDevicePushTokenAsync');
     }
diff --git a/node_modules/expo-notifications/build/getExpoPushTokenAsync.js b/node_modules/expo-notifications/build/getExpoPushTokenAsync.js
index b313f88..c795f0b 100644
--- a/node_modules/expo-notifications/build/getExpoPushTokenAsync.js
+++ b/node_modules/expo-notifications/build/getExpoPushTokenAsync.js
@@ -43,6 +43,9 @@ const productionBaseUrl = 'https://exp.host/--/api/v2/';
  * ```
  */
 export default async function getExpoPushTokenAsync(options = {}) {
+    if (Platform.OS === 'android') {
+        throw new Error("Push notifications are not supported on Android.");
+    }
     const devicePushToken = options.devicePushToken || (await getDevicePushTokenAsync());
     const deviceId = options.deviceId || (await getDeviceIdAsync());
     const projectId = options.projectId || Constants.easConfig?.projectId;
