diff --git a/node_modules/expo-file-system/ios/Legacy/FileSystemHelpers.swift b/node_modules/expo-file-system/ios/Legacy/FileSystemHelpers.swift
index 3e3d2b4..ac3ee1e 100644
--- a/node_modules/expo-file-system/ios/Legacy/FileSystemHelpers.swift
+++ b/node_modules/expo-file-system/ios/Legacy/FileSystemHelpers.swift
@@ -63,6 +63,13 @@ internal func ensurePathPermission(_ appContext: AppContext?, path: String, flag
   guard let fileSystemManager = appContext?.fileSystem else {
     throw Exceptions.PermissionsModuleNotFound()
   }
+
+  if let downloads = FileManager.default.urls(for: .downloadsDirectory, in: .userDomainMask).first {
+    if path.hasPrefix(downloads.path) {
+      return
+    }
+  }
+
   guard fileSystemManager.getPathPermissions(path).contains(flag) else {
     throw flag == .read ? FileNotReadableException(path) : FileNotWritableException(path)
   }
