diff --git a/node_modules/expo-calendar/android/src/main/java/expo/modules/calendar/CalendarModule.kt b/node_modules/expo-calendar/android/src/main/java/expo/modules/calendar/CalendarModule.kt
index 5579727..fe49b44 100644
--- a/node_modules/expo-calendar/android/src/main/java/expo/modules/calendar/CalendarModule.kt
+++ b/node_modules/expo-calendar/android/src/main/java/expo/modules/calendar/CalendarModule.kt
@@ -5,6 +5,7 @@ import android.content.ContentUris
 import android.content.ContentValues
 import android.content.Intent
 import android.database.Cursor
+import android.database.sqlite.SQLiteException
 import android.os.Bundle
 import android.provider.CalendarContract
 import android.util.Log
@@ -436,7 +437,11 @@ class CalendarModule : Module() {
       try {
         when (startDate) {
           is String -> {
-            val parsedDate = sdf.parse(startDate)
+            val parsedDate = try {
+              sdf.parse(startDate)
+            } catch (e: Exception) {
+              null
+            }
             if (parsedDate != null) {
               startCal.time = parsedDate
               calendarEventBuilder.put(CalendarContract.Events.DTSTART, startCal.timeInMillis)
@@ -626,7 +631,12 @@ class CalendarModule : Module() {
         reminderValues.put(CalendarContract.Reminders.EVENT_ID, eventID)
         reminderValues.put(CalendarContract.Reminders.MINUTES, minutes)
         reminderValues.put(CalendarContract.Reminders.METHOD, method)
-        contentResolver.insert(CalendarContract.Reminders.CONTENT_URI, reminderValues)
+        try {
+          contentResolver.insert(CalendarContract.Reminders.CONTENT_URI, reminderValues)
+        }
+        catch (e: SQLiteException) {
+
+        }
       }
     }
   }
diff --git a/node_modules/expo-calendar/expo-module.config.json b/node_modules/expo-calendar/expo-module.config.json
index d50f06d..35a0e69 100644
--- a/node_modules/expo-calendar/expo-module.config.json
+++ b/node_modules/expo-calendar/expo-module.config.json
@@ -4,12 +4,6 @@
     "modules": ["CalendarModule"]
   },
   "android": {
-    "modules": ["expo.modules.calendar.CalendarModule"],
-    "publication": {
-      "groupId": "host.exp.exponent",
-      "artifactId": "expo.modules.calendar",
-      "version": "15.0.7",
-      "repository": "local-maven-repo"
-    }
+    "modules": ["expo.modules.calendar.CalendarModule"]
   }
 }
