diff --git a/node_modules/expo/ios/AppDelegates/EXReactRootViewFactory.h b/node_modules/expo/ios/AppDelegates/EXReactRootViewFactory.h
index 2daec5d..df3fdfe 100644
--- a/node_modules/expo/ios/AppDelegates/EXReactRootViewFactory.h
+++ b/node_modules/expo/ios/AppDelegates/EXReactRootViewFactory.h
@@ -28,6 +28,7 @@ NS_SWIFT_NAME(ExpoReactRootViewFactory)
 - (UIView *)superViewWithModuleName:(NSString *)moduleName
                   initialProperties:(nullable NSDictionary *)initialProperties
                       launchOptions:(nullable NSDictionary *)launchOptions
+                bundleConfiguration:(RCTBundleConfiguration *)bundleConfiguration
                devMenuConfiguration:(nullable RCTDevMenuConfiguration *)devMenuConfiguration;
 #else
 - (UIView *)superViewWithModuleName:(NSString *)moduleName
diff --git a/node_modules/expo/ios/AppDelegates/EXReactRootViewFactory.mm b/node_modules/expo/ios/AppDelegates/EXReactRootViewFactory.mm
index 33e26db..7700901 100644
--- a/node_modules/expo/ios/AppDelegates/EXReactRootViewFactory.mm
+++ b/node_modules/expo/ios/AppDelegates/EXReactRootViewFactory.mm
@@ -27,24 +27,37 @@
 - (UIView *)viewWithModuleName:(NSString *)moduleName
              initialProperties:(nullable NSDictionary *)initialProperties
                  launchOptions:(nullable NSDictionary *)launchOptions
+           bundleConfiguration:(RCTBundleConfiguration *)bundleConfiguration
           devMenuConfiguration:(RCTDevMenuConfiguration *)devMenuConfiguration
 {
   if (self.reactDelegate != nil) {
     return [((id<EXReactDelegateProtocol>)self.reactDelegate) createReactRootViewWithModuleName:moduleName initialProperties:initialProperties launchOptions:launchOptions];
   }
-  return [super viewWithModuleName:moduleName initialProperties:initialProperties launchOptions:launchOptions devMenuConfiguration:devMenuConfiguration];
+  return [super viewWithModuleName:moduleName
+                 initialProperties:initialProperties
+                     launchOptions:launchOptions
+               bundleConfiguration:bundleConfiguration
+              devMenuConfiguration:devMenuConfiguration];
 }
 
 - (UIView *)superViewWithModuleName:(NSString *)moduleName
                   initialProperties:(nullable NSDictionary *)initialProperties
                       launchOptions:(nullable NSDictionary *)launchOptions
-               devMenuConfiguration:(nullable RCTDevMenuConfiguration *)devMenuConfiguration
+                bundleConfiguration:(RCTBundleConfiguration *)bundleConfiguration
+               devMenuConfiguration:(RCTDevMenuConfiguration *)devMenuConfiguration
 {
+  if (bundleConfiguration == nil) {
+    bundleConfiguration = [RCTBundleConfiguration defaultConfiguration];
+  }
   if (devMenuConfiguration == nil) {
     devMenuConfiguration = [RCTDevMenuConfiguration defaultConfiguration];
   }
 
-  return [super viewWithModuleName:moduleName initialProperties:initialProperties launchOptions:launchOptions devMenuConfiguration:devMenuConfiguration];
+  return [super viewWithModuleName:moduleName
+                 initialProperties:initialProperties
+                     launchOptions:launchOptions
+               bundleConfiguration:bundleConfiguration
+              devMenuConfiguration:devMenuConfiguration];
 }
 #else
 - (UIView *)viewWithModuleName:(NSString *)moduleName
diff --git a/node_modules/expo/ios/AppDelegates/ExpoReactNativeFactory.h b/node_modules/expo/ios/AppDelegates/ExpoReactNativeFactory.h
index 1facfd9..ba11f01 100644
--- a/node_modules/expo/ios/AppDelegates/ExpoReactNativeFactory.h
+++ b/node_modules/expo/ios/AppDelegates/ExpoReactNativeFactory.h
@@ -7,6 +7,6 @@
 @protocol RCTHostRuntimeDelegate;
 
 NS_SWIFT_NAME(ExpoReactNativeFactoryObjC)
-@interface EXReactNativeFactory : RCTReactNativeFactory <RCTHostDelegate, RCTHostRuntimeDelegate>
+@interface EXReactNativeFactory : RCTReactNativeFactory <RCTHostDelegate>
 
 @end
diff --git a/node_modules/expo/ios/AppDelegates/ExpoReactNativeFactory.mm b/node_modules/expo/ios/AppDelegates/ExpoReactNativeFactory.mm
index 7dab148..c6734f3 100644
--- a/node_modules/expo/ios/AppDelegates/ExpoReactNativeFactory.mm
+++ b/node_modules/expo/ios/AppDelegates/ExpoReactNativeFactory.mm
@@ -26,19 +26,6 @@
 
 #pragma mark - RCTHostDelegate
 
-// [main thread]
-- (void)hostDidStart:(nonnull RCTHost *)host
-{
-  // Setting the runtime delegate here doesn't feel right, but there is no other way
-  // to capture the `host:didInitializeRuntime:` method call.
-  // With the current API design we also depend that the runtime is initialized after the host started,
-  // which isn't obvious, especially they are invoked on different threads.
-  // Ideally if the current `RCTHostRuntimeDelegate` is part of `RCTHostDelegate`.
-  host.runtimeDelegate = self;
-}
-
-#pragma mark - RCTHostRuntimeDelegate
-
 // [JS thread]
 - (void)host:(nonnull RCTHost *)host didInitializeRuntime:(jsi::Runtime &)runtime
 {
diff --git a/node_modules/expo/ios/AppDelegates/ExpoReactNativeFactory.swift b/node_modules/expo/ios/AppDelegates/ExpoReactNativeFactory.swift
index 925e0f9..c881999 100644
--- a/node_modules/expo/ios/AppDelegates/ExpoReactNativeFactory.swift
+++ b/node_modules/expo/ios/AppDelegates/ExpoReactNativeFactory.swift
@@ -118,6 +118,7 @@ public class ExpoReactNativeFactory: ExpoReactNativeFactoryObjC, ExpoReactNative
         withModuleName: moduleName ?? defaultModuleName,
         initialProperties: initialProps,
         launchOptions: launchOptions ?? [:],
+        bundleConfiguration: RCTBundleConfiguration.default(),
         devMenuConfiguration: self.devMenuConfiguration
       )
 #else
@@ -133,7 +134,8 @@ public class ExpoReactNativeFactory: ExpoReactNativeFactoryObjC, ExpoReactNative
         withModuleName: moduleName ?? defaultModuleName,
         initialProperties: initialProps,
         launchOptions: launchOptions,
-        devMenuConfiguration: self.devMenuConfiguration
+        bundleConfiguration: RCTBundleConfiguration.default(),
+        devMenuConfiguration: self.devMenuConfiguration ?? RCTDevMenuConfiguration.default()
       )
 #else
       rootView = rootViewFactory.view(
