diff --git a/node_modules/react-native-device-info/ios/RNDeviceInfo/RNDeviceInfo.m b/node_modules/react-native-device-info/ios/RNDeviceInfo/RNDeviceInfo.m
index f27889a..05d8cbe 100644
--- a/node_modules/react-native-device-info/ios/RNDeviceInfo/RNDeviceInfo.m
+++ b/node_modules/react-native-device-info/ios/RNDeviceInfo/RNDeviceInfo.m
@@ -288,7 +288,7 @@ - (NSString *) getModel {
 }

 - (NSString *) getCarrier {
-#if (TARGET_OS_TV)
+#if (TARGET_OS_TV || TARGET_OS_MACCATALYST)
     return @"unknown";
 #else
     CTTelephonyNetworkInfo *netinfo = [[CTTelephonyNetworkInfo alloc] init];
