diff --git a/node_modules/react-native-bottom-tabs/lib/commonjs/utils/useBottomTabBarHeight.js b/node_modules/react-native-bottom-tabs/lib/commonjs/utils/useBottomTabBarHeight.js
index a1452a3..1139f51 100644
--- a/node_modules/react-native-bottom-tabs/lib/commonjs/utils/useBottomTabBarHeight.js
+++ b/node_modules/react-native-bottom-tabs/lib/commonjs/utils/useBottomTabBarHeight.js
@@ -11,7 +11,7 @@ function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e;
 function useBottomTabBarHeight() {
   const height = React.useContext(_BottomTabBarHeightContext.BottomTabBarHeightContext);
   if (height === undefined) {
-    throw new Error("Couldn't find the bottom tab bar height. Are you inside a screen in Native Bottom Tab Navigator?");
+    return 0;
   }
   return height;
 }
diff --git a/node_modules/react-native-bottom-tabs/lib/module/utils/useBottomTabBarHeight.js b/node_modules/react-native-bottom-tabs/lib/module/utils/useBottomTabBarHeight.js
index ce2520a..32d3bfb 100644
--- a/node_modules/react-native-bottom-tabs/lib/module/utils/useBottomTabBarHeight.js
+++ b/node_modules/react-native-bottom-tabs/lib/module/utils/useBottomTabBarHeight.js
@@ -5,7 +5,7 @@ import { BottomTabBarHeightContext } from "./BottomTabBarHeightContext.js";
 export function useBottomTabBarHeight() {
   const height = React.useContext(BottomTabBarHeightContext);
   if (height === undefined) {
-    throw new Error("Couldn't find the bottom tab bar height. Are you inside a screen in Native Bottom Tab Navigator?");
+    return 0;
   }
   return height;
 }
diff --git a/node_modules/react-native-bottom-tabs/src/utils/useBottomTabBarHeight.ts b/node_modules/react-native-bottom-tabs/src/utils/useBottomTabBarHeight.ts
index 90eb6b5..a29f179 100644
--- a/node_modules/react-native-bottom-tabs/src/utils/useBottomTabBarHeight.ts
+++ b/node_modules/react-native-bottom-tabs/src/utils/useBottomTabBarHeight.ts
@@ -6,9 +6,7 @@ export function useBottomTabBarHeight() {
   const height = React.useContext(BottomTabBarHeightContext);
 
   if (height === undefined) {
-    throw new Error(
-      "Couldn't find the bottom tab bar height. Are you inside a screen in Native Bottom Tab Navigator?"
-    );
+    return 0;
   }
 
   return height;
