diff --git a/node_modules/react-native-vector-icons/RNVectorIcons.podspec b/node_modules/react-native-vector-icons/RNVectorIcons.podspec
index 5877f53..ec19a9c 100644
--- a/node_modules/react-native-vector-icons/RNVectorIcons.podspec
+++ b/node_modules/react-native-vector-icons/RNVectorIcons.podspec
@@ -13,7 +13,7 @@ Pod::Spec.new do |s|
   s.source         = { :git => package["repository"]["url"], :tag => "v#{s.version}" }
 
   s.source_files   = 'RNVectorIconsManager/**/*.{h,m,mm,swift}'
-  s.resources      = "Fonts/*.ttf"
+  s.resources      = "Fonts/MaterialCommunityIcons.ttf", "Fonts/MaterialIcons.ttf"
   s.preserve_paths = "**/*.js"
   # React Native Core dependency
   if defined? install_modules_dependencies
