diff --git a/node_modules/react-native-navigation/lib/ios/RNNSplitViewControllerPresenter.m b/node_modules/react-native-navigation/lib/ios/RNNSplitViewControllerPresenter.m
index a1d9349..5143930 100644
--- a/node_modules/react-native-navigation/lib/ios/RNNSplitViewControllerPresenter.m
+++ b/node_modules/react-native-navigation/lib/ios/RNNSplitViewControllerPresenter.m
@@ -18,6 +18,11 @@ - (void)applyOptions:(RNNNavigationOptions *)options {
 	[self.splitViewController rnn_setMaxWidth:options.splitView.maxWidth];
 	[self.splitViewController rnn_setPrimaryBackgroundStyle:options.splitView.primaryBackgroundStyle];
 
+    if (@available(iOS 13, *)) {
+        [[self.splitViewController view] setBackgroundColor:[UIColor separatorColor]];
+    } else {
+        [[self.splitViewController view] setBackgroundColor:[UIColor colorWithRed:0.0f green:0.0f blue:0.0f alpha:0.1f]];
+    }
 }
 
 
