diff --git a/node_modules/react-native-vector-icons/RNVectorIcons.podspec b/node_modules/react-native-vector-icons/RNVectorIcons.podspec
index 4b716ee..d0764c7 100644
--- a/node_modules/react-native-vector-icons/RNVectorIcons.podspec
+++ b/node_modules/react-native-vector-icons/RNVectorIcons.podspec
@@ -12,7 +12,7 @@ Pod::Spec.new do |s|
   s.platforms      = { :ios => "9.0", :tvos => "9.0" }
   s.source         = { :git => "https://github.com/oblador/react-native-vector-icons.git", :tag => "v#{s.version}" }
   s.source_files   = 'RNVectorIconsManager/**/*.{h,m}'
-  s.resources      = "Fonts/*.ttf"
+  s.resources      = "Fonts/MaterialCommunityIcons.ttf", "Fonts/MaterialIcons.ttf"
   s.preserve_paths = "**/*.js"
   s.dependency 'React-Core'

diff --git a/node_modules/react-native-vector-icons/fonts.gradle b/node_modules/react-native-vector-icons/fonts.gradle
index 5485fac..40382cc 100644
--- a/node_modules/react-native-vector-icons/fonts.gradle
+++ b/node_modules/react-native-vector-icons/fonts.gradle
@@ -28,5 +28,10 @@ afterEvaluate {
         def targetName = variant.name.capitalize()
         def generateAssetsTask = tasks.findByName("generate${targetName}Assets")
         generateAssetsTask.dependsOn(fontCopyTask)
+
+        def lintVitalAnalyzeTask = tasks.findByName("lintVitalAnalyzeRelease")
+        if (lintVitalAnalyzeTask != null) {
+            lintVitalAnalyzeTask.dependsOn(fontCopyTask)
+        }
     }
 }
