diff --git a/node_modules/react-native-bottom-tabs/ios/RCTTabViewComponentView.mm b/node_modules/react-native-bottom-tabs/ios/RCTTabViewComponentView.mm
index 09ab604b..93e9885f 100644
--- a/node_modules/react-native-bottom-tabs/ios/RCTTabViewComponentView.mm
+++ b/node_modules/react-native-bottom-tabs/ios/RCTTabViewComponentView.mm
@@ -18,7 +18,7 @@
 #import <React/RCTImageSource.h>
 #import <React/RCTBridge+Private.h>
 #import "RCTImagePrimitivesConversions.h"
-#import "RCTConversions.h"
+#import <React/RCTConversions.h>
 #import <react/utils/ManagedObjectWrapper.h>
 
 #if TARGET_OS_OSX
