diff --git a/node_modules/react-native-paper/src/components/IconButton/IconButton.tsx b/node_modules/react-native-paper/src/components/IconButton/IconButton.tsx
index 8809882..c7aae22 100644
--- a/node_modules/react-native-paper/src/components/IconButton/IconButton.tsx
+++ b/node_modules/react-native-paper/src/components/IconButton/IconButton.tsx
@@ -146,7 +146,7 @@ const IconButton = React.forwardRef<View, Props>(
         customContainerColor,
       });
 
-    const buttonSize = isV3 ? size + 2 * PADDING : size * 1.5;
+    const buttonSize = size * 1.5;
 
     const borderStyles = {
       borderWidth: isV3 && mode === 'outlined' && !selected ? 1 : 0,
diff --git a/node_modules/react-native-paper/src/components/Typography/v2/Caption.tsx b/node_modules/react-native-paper/src/components/Typography/v2/Caption.tsx
index 135e36a..48e122f 100644
--- a/node_modules/react-native-paper/src/components/Typography/v2/Caption.tsx
+++ b/node_modules/react-native-paper/src/components/Typography/v2/Caption.tsx
@@ -45,6 +45,5 @@ const styles = StyleSheet.create({
     fontSize: 12,
     lineHeight: 20,
     marginVertical: 2,
-    letterSpacing: 0.4,
   },
 });
diff --git a/node_modules/react-native-paper/src/components/Typography/v2/Paragraph.tsx b/node_modules/react-native-paper/src/components/Typography/v2/Paragraph.tsx
index 13b7d62..dec8598 100644
--- a/node_modules/react-native-paper/src/components/Typography/v2/Paragraph.tsx
+++ b/node_modules/react-native-paper/src/components/Typography/v2/Paragraph.tsx
@@ -44,6 +44,5 @@ const styles = StyleSheet.create({
     fontSize: 14,
     lineHeight: 20,
     marginVertical: 2,
-    letterSpacing: 0.25,
   },
 });
diff --git a/node_modules/react-native-paper/src/components/Typography/v2/Subheading.tsx b/node_modules/react-native-paper/src/components/Typography/v2/Subheading.tsx
index 37a1cad..aba08f8 100644
--- a/node_modules/react-native-paper/src/components/Typography/v2/Subheading.tsx
+++ b/node_modules/react-native-paper/src/components/Typography/v2/Subheading.tsx
@@ -45,6 +45,5 @@ const styles = StyleSheet.create({
     fontSize: 16,
     lineHeight: 24,
     marginVertical: 2,
-    letterSpacing: 0.5,
   },
 });
diff --git a/node_modules/react-native-paper/src/components/Typography/v2/Title.tsx b/node_modules/react-native-paper/src/components/Typography/v2/Title.tsx
index d8592b4..b110414 100644
--- a/node_modules/react-native-paper/src/components/Typography/v2/Title.tsx
+++ b/node_modules/react-native-paper/src/components/Typography/v2/Title.tsx
@@ -44,6 +44,5 @@ const styles = StyleSheet.create({
     fontSize: 20,
     lineHeight: 30,
     marginVertical: 2,
-    letterSpacing: 0.15,
   },
 });
