diff --git a/node_modules/react-native/scripts/cocoapods/utils.rb b/node_modules/react-native/scripts/cocoapods/utils.rb
index 718304e..54ea36c 100644
--- a/node_modules/react-native/scripts/cocoapods/utils.rb
+++ b/node_modules/react-native/scripts/cocoapods/utils.rb
@@ -560,7 +560,8 @@ class ReactNativePodsUtils
                     "NSAppTransportSecurity" => ats_configs
                 }
             else
-                plist["NSAppTransportSecurity"] = ats_configs
+                plist["NSAppTransportSecurity"] ||= {}
+                plist["NSAppTransportSecurity"] = plist["NSAppTransportSecurity"].merge(ats_configs)
             end
             Xcodeproj::Plist.write_to_path(plist, fullPlistPath)
         end
