diff --git a/node_modules/react-native-share/android/src/main/java/cl/json/social/TargetChosenReceiver.java b/node_modules/react-native-share/android/src/main/java/cl/json/social/TargetChosenReceiver.java
index 65889d0..00cf07c 100644
--- a/node_modules/react-native-share/android/src/main/java/cl/json/social/TargetChosenReceiver.java
+++ b/node_modules/react-native-share/android/src/main/java/cl/json/social/TargetChosenReceiver.java
@@ -46,7 +46,7 @@ public class TargetChosenReceiver extends BroadcastReceiver {
                 context.unregisterReceiver(sLastRegisteredReceiver);
             }
             sLastRegisteredReceiver = new TargetChosenReceiver();
-            context.registerReceiver(sLastRegisteredReceiver, new IntentFilter(sTargetChosenReceiveAction));
+            context.registerReceiver(sLastRegisteredReceiver, new IntentFilter(sTargetChosenReceiveAction), context.RECEIVER_NOT_EXPORTED);
         }

         Intent intent = new Intent(sTargetChosenReceiveAction);
