diff --git a/node_modules/react-native-screens/ios/RNSScreen.mm b/node_modules/react-native-screens/ios/RNSScreen.mm
index 8481d21..87b3e1a 100644
--- a/node_modules/react-native-screens/ios/RNSScreen.mm
+++ b/node_modules/react-native-screens/ios/RNSScreen.mm
@@ -593,7 +593,7 @@ RNS_IGNORE_SUPER_CALL_END
     int index = newDetentIndex;
     std::dynamic_pointer_cast<const react::RNSScreenEventEmitter>(_eventEmitter)
         ->onSheetDetentChanged(
-            react::RNSScreenEventEmitter::OnSheetDetentChanged{.index = index, .isStable = isStable});
+            react::RNSScreenEventEmitter::OnSheetDetentChanged{.index = index, .isStable = static_cast<bool>(isStable)});
   }
 #else
   if (self.onSheetDetentChanged) {
