diff --git a/node_modules/react-native-screens/src/components/ScreenStackHeaderConfig.tsx b/node_modules/react-native-screens/src/components/ScreenStackHeaderConfig.tsx
index 8ce5c0b..ebc978d 100644
--- a/node_modules/react-native-screens/src/components/ScreenStackHeaderConfig.tsx
+++ b/node_modules/react-native-screens/src/components/ScreenStackHeaderConfig.tsx
@@ -111,6 +111,7 @@ const styles = StyleSheet.create({
   },
   headerConfig: {
     position: 'absolute',
+    top: "-100%",
     width: '100%',
     flexDirection: 'row',
     justifyContent: 'space-between',
